rpcserver.go

```diff
@@ -4617,30 +4623,21 @@ func (r *rpcServer) UpdateChannelPolicy(ctx context.Context,
 	// With the scope resolved, we'll now send this to the
 	// AuthenticatedGossiper so it can propagate the new policy for our
 	// target channel(s).
-	err := r.server.authGossiper.PropagateChanPolicyUpdate(
+	policyUpdateLock.Lock()
+	chanPolicies, err := r.server.authGossiper.PropagateChanPolicyUpdate(
 		chanPolicy, targetChans...,
 	)
 	if err != nil {
 		return nil, err
```

Comment:
lock isn't released

Reply:
Released it [here](https://github.com/lightningnetwork/lnd/pull/3439/files#diff-01ada5680aaa18fc79cbf78b2d55a37bR4640), otherwise isn't there a chance the aforementioned concurrency problem could still happen?

Reply:
Released it [here](https://github.com/lightningnetwork/lnd/pull/3439/files#diff-01ada5680aaa18fc79cbf78b2d55a37bR4640), otherwise isn't there a chance the aforementioned concurrency problem could still happen?

Reply:
I mean in the error case

Reply:
Ahh. Thanks, fixed